### ISSUE TITLE
Low-hanging fruit: improve Nyxt stylesheets

### ISSUE THREAD

**Issue by aartaka** [atlas-engineer/nyxt#2188]
Right now out CSS is quite unever between pages -- some buttons change color when hovered, some don't; some links are underlined, some aren't; some lists have default markers, some have custom markers, some have no markers at all. We need to make styling of Nyxt-specific interfaces clear and pretty. Notably:
- Make all the interactive elements hoverable, and change the element state on hover.
- Ensure proper color contrast of all the element states.
- Ensure that similar styles are abstracted into `buffer`'s `style` slot to not clutter modes with lookalike styles.
- Stick to ARIA and semantic HTML guidelines (e.g. use `alt` text for images, provide `aria-role` for atypical interfaces, add tooltips to buttons...)

Help will be much appriciated! Should not be too difficult too, given that it's mostly a simple work with CSS :)

**Comment by aartaka** [comment]
More pointers: [Material Design Theming Guidelines](https://material.io/design/material-theming/implementing-your-theme.html#shape)

**Comment by jmercouris** [comment]
Consider that we are pretty picky :-D but also quite open!

**Comment by aadcg** [comment]
@aartaka does it still make sense to keep this one open?

**Comment by aartaka** [comment]
> @aartaka does it still make sense to keep this one open?

Yes, our styling is still somewhat inconsistent and would benefit from some polish.

**Comment by aadcg** [comment]
I feel that this issue is too broad and hard to extract actionable tasks. Can we re-think it? Also, we now have @lansingthomas.

**Comment by jmercouris** [comment]
Agreed, too abstract.

**Comment by aadcg** [comment]
@aartaka could you close it? Thanks.

**Comment by aartaka** [comment]
Actually, let's make it more concrete: 
- We need use action responsiveness for interactive elements: https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes#user_action_pseudo-classes
- We need location management for links and sections: https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes#user_action_pseudo-classes
- And we need some input styling for our inputs: https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes#location_pseudo-classes

**Comment by aartaka** [comment]
I'm modifying the toplevel comment with these.
